packages/blockchain/algo/src/lib/services/algo.tx.ts

```diff
@@ -352,6 +358,107 @@ export const algoTxService = (args: { algoWeb: AlgoWeb }, apiCalls: AlgoApiCalls
     return Buffer.from(signedTxn).toString('hex')
   }
 
+  const prepareCreateFractionalNFTSignedTransaction = async (
+    body: MintMultiToken | MintMultiTokenKMS,
+    testnet = false,
+    algoWeb: AlgoWeb,
+    provider?: string,
+  ) => {
+    const algodClient = algoWeb.getClient(testnet, provider)
+    const params = await algodClient.getTransactionParams().do()
+
+    const decoder = new base32.Decoder({ type: 'rfc4648' })
+
+    const from = isWithSignatureId(body)
+      ? body.from
+      : algoWallet().generateAddressFromPrivatetKey((body as MintMultiToken).fromPrivateKey)
+
+    const v = Math.floor(Math.log10(new BigNumber(body.amount).toNumber()))
+    const txn = algosdk.makeAssetCreateTxnWithSuggestedParams(
+      from,
+      undefined,
+      new BigNumber(10).pow(v).toNumber(),
+      v,
+      false,
+      from,
+      undefined,
+      undefined,
+      undefined,
+      body.tokenId,
+      body.contractAddress,
+      undefined,
+      undefined,
+      params,
+    )
+    if ('signatureId' in body) {
+      return JSON.stringify(txn)
+    }
+    const secretKey = new Uint8Array(decoder.write(body.fromPrivateKey).buf)
+    return Buffer.from(txn.signTxn(secretKey)).toString('hex')
+  }
+
+  const prepareTransferFractionalNFTSignedTransaction = async (
+    body: TransferMultiToken | TransferMultiTokenKMS,
+    testnet = false,
+    algoWeb: AlgoWeb,
+    provider?: string,
+  ) => {
+    const algodClient = algoWeb.getClient(testnet, provider)
+    const params = await algodClient.getTransactionParams().do()
+
+    const decoder = new base32.Decoder({ type: 'rfc4648' })
+
+    const from = isWithSignatureId(body)
+      ? body.from
+      : algoWallet().generateAddressFromPrivatetKey((body as TransferMultiToken).fromPrivateKey)
+
+    const txn = algosdk.makeAssetTransferTxnWithSuggestedParams(
+      from,
+      body.to,
+      undefined,
+      undefined,
+      new BigNumber(body.amount).toNumber(),
+      undefined,
+      new BigNumber(body.contractAddress).toNumber(),
+      params,
+      undefined,
+    )
+    if ('signatureId' in body) {
```

Comment:
either use isWithSignatureId(body) or 'signatureId' in body -> make the code consistent